src/main/java/de/tum/in/www1/artemis/service/connectors/GitService.java

```diff
@@ -621,22 +621,48 @@ public PullResult pull(Repository repo) throws GitAPIException {
     }
 
     /**
-     * Hard reset local repository to origin/master.
+     * Get branch that origin/HEAD points to, useful to handle default branches that are not master
      *
      * @param repo Local Repository Object.
+     * @return name of the origin/HEAD branch, e.g. 'main' or null if there is no HEAD
      */
-    public void resetToOriginMaster(Repository repo) {
+    public String getOriginHead(Repository repo) throws GitAPIException {
+        Git git = new Git(repo);
+        var originHeadRef = git.lsRemote().callAsMap().get("HEAD");
```

Comment:
Is there no method to retrieve the head directly?

Reply:
https://www.vogella.com/tutorials/JGit/article.html uses at least `Constants.HEAD`

Reply:
To get the same behavior as before with `origin/master`, we would want to reset to `origin/<default_branch>`. Usually `git clone` sets the symbolic ref `origin/HEAD` to the default branch it encountered during the clone. At least in the default configuration JGit does not do that, so we can't use `repo.resolve("origin/HEAD")`. Because we sometimes checkout a different branch ([example](https://github.com/ls1intum/Artemis/blob/a59b08bbf610da6bb2acd28b12a3f5951f2a91ac/src/main/java/de/tum/in/www1/artemis/service/connectors/GitService.java#L739)), we can't use our local `HEAD` directly.